src/databases.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the Databases class to handle database relations and state."""
+
+import logging
+import pathlib
+import typing
+
+import ops.framework
+import yaml
+from charms.data_platform_libs.v0.data_interfaces import DatabaseRequires, DatabaseRequiresEvent
+from ops.charm import CharmBase
+
+from constants import FLASK_CONTAINER_NAME, FLASK_DATABASE_NAME, FLASK_SERVICE_NAME
+from exceptions import InvalidDatabaseRelationDataError, PebbleNotReadyError
+
+logger = logging.getLogger(__name__)
+
+
+# We need to derive from ops.framework.Object to subscribe to callbacks
+# from ops.framework. See: https://github.com/canonical/operator/blob/main/ops/framework.py#L782
+class Databases(ops.framework.Object):  # pylint: disable=too-few-public-methods
+    """A class handling databases relations and state.
+
+    Attrs:
+        _charm: The main charm. Used for events callbacks
+        _databases: A dict of DatabaseRequires to store relations
+    """
+
+    def __init__(self, charm: CharmBase):
+        """Initialize a new instance of the Databases class.
+
+        Args:
+            charm: The main charm. Used for events callbacks
+        """
+        # The following is necessary to be able to subscribe to callbacks from ops.framework
+        super().__init__(charm, "databases")
+        self._charm = charm
+
+        metadata = yaml.safe_load(pathlib.Path("metadata.yaml").read_text(encoding="utf-8"))
+        self._db_interfaces = (
+            name for name in list(metadata["requires"]) if name.startswith("db_")
```

Comment:
Instead of looking for a prefix I'd rather look for requires that uses `postgres_client` or `mysql_client` as interface, which would make it more generic

Reply:
I understand. Is it not possible that we use those interfaces for something else than the "db" ?
It's probably too much of an edge case now that I think of it but I thought that we could use postgres/mysql for something cache related or something else. In that case we could have those interfaces used in another way than intended.

Reply:
Whatever the reason is, everytime we have a postgres_client or mysql_client interface used, we need to instanciate the requirer to manage the relation.
That's why I'm suggesting this change

Reply:
Fixed